crates/kas-widgets/src/stack.rs

```diff
@@ -42,19 +43,32 @@ impl_scope! {
     /// By default, all pages are configured and sized. To avoid configuring
     /// hidden pages (thus preventing these pages from affecting size)
     /// call [`Self::set_size_limit`] or [`Self::with_size_limit`].
-    #[impl_default]
     #[derive(Clone, Debug)]
     #[widget]
     pub struct Stack<W: Widget> {
         core: widget_core!(),
         align_hints: AlignHints,
         widgets: Vec<(W, State)>,
         active: usize,
-        size_limit: usize = usize::MAX,
+        size_limit: usize,
         next: usize,
         id_map: HashMap<usize, usize>, // map key of Id to index
     }
 
+    impl Default for Self {
+        fn default() -> Self {
+            Stack {
+                core: Default::default(),
+                align_hints: AlignHints::NONE,
+                widgets: Vec::new(),
+                active: 0,
+                size_limit: usize::MAX,
+                next: 0,
+                id_map: HashMap::new(),
+            }
+        }
+    }
+
```

Comment:
This is the only case where I opted to replace `#[impl_default]` with a hand-written `Default` impl. Why:

1. This is a widget definition with a significant amount of code, therefore more important than most other uses of `#[impl_default]` to be automatically formattable.
2. `#[derive(Default)]` would use the wrong generic bounds here, so RFC 3681 is not sufficient to replace this usage of `#[impl_default]`